### PR TITLE
Use core-js URL polyfill instead of whatwg-url

### PR DESCRIPTION
**Pull request by felixfbecker**
whatwg-url has a very large bundle size, because it is basically the pure-JS reference implementation, including a lot of details with domain encoding, punycode, etc. It's not intended as polyfill.

This reduces our bundle size by 28% (0.692MB -> 0.499MB, Δ ~200KB)

Verified that core-js `URL` parses out `hostname`s properly from custom hosts (`git://`), which is the primary thing we care about.

### CODE REVIEW

File: shared/src/polyfills.ts [hunk 1]
```diff
@@ -1,14 +0,0 @@
-// Commonly needed by extensions (used by vscode-jsonrpc)
-import 'core-js/web/immediate'
-
-import 'symbol-observable'
-
-// This gets automatically expanded into imports that only pick what we need.
-import 'core-js/stable'
-import 'regenerator-runtime/runtime'
```
Comment (eseliger) [inline shared/src/polyfills.ts:8]: this one is not needed anymore? Can't find it anywhere else

Comment (eseliger) [reply shared/src/polyfills.ts:8]: If so, can the dependency be removed?

File: shared/src/polyfills.ts [hunk 2]
```diff
@@ -1,14 +0,0 @@
...
-
-// Polyfill URL because Chrome and Firefox are not spec-compliant
-// Hostnames of URIs with custom schemes (e.g. git) are not parsed out
-import { URL, URLSearchParams } from 'whatwg-url'
-Object.assign(URL, self.URL) // keep static methods like URL.createObjectURL()
-Object.assign(self, { URL, URLSearchParams })
```
Comment (eseliger) [inline shared/src/polyfills.ts:14]: Those are covered by core-js's implementation?

**Comment by felixfbecker** [comment]
I verified that core-js `URL` parses out `hostname`s properly from custom hosts (`git://`), which is the primary thing we care about. Please let me know if you can think of another edge case!

File: shared/src/polyfills.ts (hunk 2 above)
Comment (felixfbecker) [reply shared/src/polyfills.ts:14]: Yep

File: shared/src/polyfills.ts (hunk 1 above)
Comment (felixfbecker) [reply shared/src/polyfills.ts:8]: Oops, that was accidental. It didn't lead to an error though, which leads me to believe all the browsers we target in our browserslist support generators now, and we can indeed remove the dependency.

Comment (felixfbecker) [reply shared/src/polyfills.ts:8]: Removed

**Pull request merged**